src/ui/components/ScrollViewRefetch.tsx

```diff
@@ -12,8 +13,13 @@ type Props = Omit<ScrollViewProps, 'refreshControl'> & {
 
 export function ScrollViewRefetch({children, refetchQueries, ...props}: Props) {
   const {colors} = useTheme();
+  const {isConnected} = useNetInfo();
   const {refreshing, refetch} = useRefetch(refetchQueries);
 
+  useEffect(() => {
+    refetch();
+  }, [isConnected, refetch]);
+
```

Comment:
This would be an incorrect place to implement this behaviour. The `ScrollViewRefetch` component is only supposed to handle the pull to refresh only. New responsibilities are added to the component which are unrelated.

Another approach can look something like following.

```
function useApiRefetch() {
    const {isConnected} = useNetInfo();
    const {refetch} = useRefetch();

    useEffect(() => {
      if(isConnected){
        refetch();
      }
    }, [isConnected, refetch]);
}
```
And use this `useApiRefetch()` hook somewhere high up in the component chain. such as in the `AppNavigator` component.